docs/src/references/c8y-configuration-management.md

```diff
@@ -264,4 +271,75 @@ When there is no coresponding config file on the thin-edge device the plugin sen
+    * On c8y read request for a configuration, `c8y_configuration_plugin` returns the `current` file content.
+    * On c8y write request for a configuration, `c8y_configuration_plugin` updates the `desired` file content and waits for the `current` content to be updated to return the operation status to cumulocity.  
+    * The current behavior of c8y_configuration_plugin can be simulated with desired = current = path when only a path is provided.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { type = 'bar.conf', child_id = 'child1', current = '/http-root/config/current/child1/bar.conf', desired = '/http-root/config/desired/child1/bar.conf' } }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device child1: starts
+  * external device child1: sends all its config files with HTTP PUT requests to the thin-edge device; 
+
+    Example: HTTP PUT to `http://<thin-edge IP address>/tedge/filetransfer/current/child1/bar.conf`
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: sends the file `/http-root/config/current/child1/bar.conf` to C8Y<br/><br/>
+    NOTE: The responsibility to assure that the latest config file content on the thin-edge device is always on the external device. When there is no file the plugin sends an error to the cloud on an every incoming config retrieval request.
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: stores the files received from C8Y to `/http-root/config/desired/child1/bar.conf`
+  * C8Y config plugin: notifies the external device about the new configuration via MQTT (reference to section Notifications above)
+  * external device child1: recognizes the MQTT notification and downloads the new config with an HTTP GET request
+  
+    Example: HTTP GET to `http://<thin-edge IP address>/tedge/filetransfer/config/desired/child1/bar.conf`
+  * external device child1: after processing, it updates the _current_ configuration on the thin-edge device with an HTTP PUT request.
+
+    Example: HTTP GET to `http://<thin-edge IP address>/tedge/filetransfer/config/current/child1/bar.conf`
+  * C8Y config plugin: recognizes the MQTT notification about the updated file in the `current` folder,<br/>
+    compares `/http-root/config/desired/child1/bar.conf` and `/http-root/config/current/child1/bar.conf`.
+    * if both files are equal, the plugin reports success to C8Y.
+    * if files are not equal, the plugin reports an error to C8Y.
+    * TO-BE-DEFINED:
+      * Which kind of error to report to C8Y in case of unequal files?
+      * Any timeout to wait for update in folder `current`?
```

Comment:
Another open question is:

* Do we provide an API for a child device to notify on configuration errors.

Reply:
Yes, we provide that.

In happy case a child-device responses to the plugin via MQTT:

   * Topic: `tedge/<childid>/commands/res/successful/config_update/<config file path>`
   * Payload: EMPTY

In fail case a child-device responses to the plugin via MQTT:

   * Topic: `tedge/<childid>/commands/res/failed/config_update/<config file path>`
   * Payload: fail reason as string
   
 See also https://github.com/thin-edge/thin-edge.io/blob/ad15003854c59df59a41d30ebaae2a04d661596f/docs/src/references/c8y-configuration-management.md#pushing-configuration-file-update-to-child-device-from-cloud